controllers/warehousesController.js

```diff
@@ -0,0 +1,108 @@
+const Joi = require("joi");
+const knex = require("knex")(require("../knexfile"));
+
+// {
+//     "warehouse_id": 1,
+//     "item_name": "Paper Towels",
+//     "description": "Made out of military-grade synthetic materials, these paper towels are highly flammable, yet water resistant, and easy to clean.",
+//     "category": "Gear",
+//     "status": "Out of Stock",
+//     "quantity": 0
+//   }
+
+// POST -- Create New InveNtory of Warehouse by its ID
+const createNewWarehouse = async (req, res) => {
+  try {
+    // inventorySchema for req body validations
+    const warehouseSchema = Joi.object({
+      warehouse_name: Joi.string().required(),
+      address: Joi.string().required(),
+      contact_name: Joi.string().required(),
+      contact_information: Joi.string().required(),
+      actions: Joi.string().required().required(),
+    });
+
+    const { error } = warehouseSchema.validate(
+      //convert status from REQ BODY to lowerCase to pass Validation
+      { ...req.body, status: req.body.status.toLowerCase() },
+      {
+        abortEarly: false,
+      }
+    );
+
+    // Handle validation errors
+    if (error) {
+      return res.status(400).send({
+        message: error.details.map((detail) => detail.message).join(", "),
+      });
+    }
+    // check if  warehouse_id all ready exist in warehouse
+    const { warehouse_id, item_name, description, category, status, quantity } =
+      req.body;
+
+    const warehouseAllReadyExist = await knex("warehouses").where(
+      "id",
+      warehouse_id
+    );
+    if (warehouseAllReadyExist.length === 0)
+      return res.status(400).send({
+        message:
+          "The warehouse you're trying to add inventory to does not exist. Please add the warehouse first.",
```

Comment:
this line needs to be updated to  "The warehouse you're trying to add inventory to does exist ,  Please add a different 
 warehouse first."